pkgs/servers/web-apps/searxng/default.nix

```diff
@@ -3,24 +3,40 @@
 , fetchFromGitHub
 }:
 
+let
+  commit = "a446dea1bb492eac417de9a900fae7cdf94aeec0";
+in
 python3.pkgs.buildPythonApplication rec {
   pname = "searxng";
-  version = "unstable-2023-06-26";
+  version = "unstable-2023-07-19";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
-    rev = "da7c30291dcf53cc5b3d98f9aada5615cd1593a9";
-    sha256 = "sha256-kbNw/YgcBZNkmn2nmsnEnc9Y8MJg3zGFdW1x9GIo+dM=";
+    rev = commit;
+    sha256 = "sha256-iZDaKCkDlp3O3IixWdXVykNRIxas+irG0dWAOU4wycI=";
   };
 
   postPatch = ''
     sed -i 's/==.*$//' requirements.txt
   '';
 
-  preBuild = ''
-    export SEARX_DEBUG="true";
-  '';
+  preBuild =
+    let
+      versionString = lib.concatStringsSep "." (builtins.tail (lib.splitString "-" version));
+      commitAbbrev = builtins.substring 0 8 commit;
+    in
+    ''
+      export SEARX_DEBUG="true";
+
+      cat > searx/version_frozen.py <<EOF
+      VERSION_STRING="${versionString}+${commitAbbrev}"
+      VERSION_TAG="${versionString}+${commitAbbrev}"
```

Comment:
Is `${versionString}` replaced by the nixos package version (`unstable-2023-07-19` from above)?

If so you might run into trouble with [PEP-440 Version scheme](https://peps.python.org/pep-0440/#version-scheme) .. in the past we (SearXNG team) we had some trouble with the version scheme accepted by pip and other versioning tools (that implement strict PEP-440' versions scheme).

- https://github.com/searxng/searxng/pull/2122
- https://github.com/searxng/searxng/pull/2117

Reply:
`versionString` is calculated from the package version in line 26, so `versionString = lib.concatStringsSep "." (builtins.tail (lib.splitString "-" version))`. Currently, this results in `2023.07.19`.

I don't think PEP-440 impacts the nix package at all as versioning is entirely handled through nix (which only cares about the package version). The `VERSION_STRING` written to `version_frozen.py` is only used for displaying in the web interface as far as I can tell.

Reply:
That should be fine, as long as pip isn't complaining. If we need more information than currently available we should just fill them with dummy values to keep things simple.